ldk/javascript/src/whisper/index.ts

```diff
@@ -264,6 +284,33 @@ export interface WhisperAptitude {
   create(whisper: NewWhisper): Promise<Whisper>;
 }
 
-export function create(whisper: NewWhisper): Promise<Whisper> {
-  return promisifyWithParam(whisper, oliveHelps.whisper.create);
+function addUpdateHandler(whisper: OliveHelps.Whisper): Whisper {
+  const updateFunc = whisper.update;
+  const externalWhisper: Whisper = mapToExternalWhisper(whisper);
+  externalWhisper.update = (updateWhisper: UpdateWhisper, cb?: (err: Error) => void) => {
+    updateFunc(mapToInternalUpdateWhisper(updateWhisper), cb);
+  };
+  return externalWhisper;
 }
+
+export function create(whisper: NewWhisper): Promise<Whisper> {
+  const internalWhisper: OliveHelps.NewWhisper = mapToInternalWhisper(whisper);
+
+  return new Promise((resolve, reject) => {
```

Comment:
I think after moving addUpdateHandler to mapper you should be able to use: 'promisifyMappedWithParam'.